### PR TITLE
Web3 settings section

### PR DESCRIPTION
**Pull request by supermassive**
Resolves https://github.com/brave/brave-browser/issues/14516

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
- Setting prefs in new locations should work.
- This https://github.com/brave/brave-browser/issues/25383 should work(now just hides wallet card)
- Navigations to `Wallet networks` should work
- Navigations to `Ipfs peers` and `Ipns keys` should work
- Navigations to settings/wallet or settings/wallet/networks from wallet ui should work. Now navigates directly to wallet section as before which seems ok.

https://user-images.githubusercontent.com/168356/206421150-69d2bb30-2fb8-49cc-8f06-272696773910.mp4

### CODE REVIEW

**Review by goodov — APPROVED** [review]

**Review by yrliou — APPROVED** [review]

File: browser/resources/settings/brave_overrides/basic_page.ts [hunk 1]
```diff
@@ -258,40 +275,43 @@ RegisterPolymerTemplateModifications({
...
-      sectionSearch.insertAdjacentElement('afterend', sectionExtensions)
-      // Insert Wallet and IPFS
-      if (isBraveWalletAllowed) {
-        sectionExtensions.insertAdjacentElement('afterend', sectionWallet)
-        sectionWallet.insertAdjacentElement('afterend', sectionIPFS)
-      } else {
-        sectionExtensions.insertAdjacentElement('afterend', sectionIPFS)
+      last = last.insertAdjacentElement('afterend', sectionExtensions)
+      // Insert Wallet
+      if (sectionWallet) {
```
Comment (petemill) [inline browser/resources/settings/brave_overrides/basic_page.ts:305]: this is a nice improvement

File: browser/resources/settings/brave_overrides/settings_menu.ts [hunk 2]
```diff
@@ -297,29 +297,15 @@ RegisterPolymerTemplateModifications({
...
-        'brave_settings:wallet',
-        'wallet',
-      )
-      extensionEl.insertAdjacentElement('afterend', walletEl)
-    }
 
-    const ipfsEl = createMenuElement(
-      loadTimeData.getString('braveIPFS'),
-      '/ipfs',
-      'brave_settings:ipfs',
```
Comment (petemill) [inline browser/resources/settings/brave_overrides/settings_menu.ts:315]: I think this means we should remove the ipfs icon from brave_settings unless we show it anywhere else?

**Review by petemill — APPROVED** [review]
Nicely done. Please check for the icon removal before merge.

File: browser/resources/settings/brave_overrides/settings_menu.ts (hunk 2 above)
Comment (supermassive) [reply browser/resources/settings/brave_overrides/settings_menu.ts:315]: https://github.com/brave/brave-core/pull/16290/commits/46486a1a3ed9dde0cb211de0cc787d5d5b15ac86

**Pull request merged**